### PR TITLE
Add readme for secret manager

### PR DESCRIPTION
**Pull request by alexperovich**

### CODE REVIEW

**Review by jakubstilec — APPROVED** [review]

File: src/Microsoft.DncEng.SecretManager/Readme.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  environment: staging or production
+```
+
+### Text
+This type should be used sparingly, and only for things that aren't actually secret.
+```yaml
+type: text
+parameters:
+  description: what the text is
+```
```
Comment (jakubstilec) [inline src/Microsoft.DncEng.SecretManager/Readme.md:196]: There should be one more block for domain account:

 ### Domain Account
```yaml
type: domain-account
parameters:
  domain: domain name
  user: user account name
  description: additional description for rotation of the password
```

File: src/Microsoft.DncEng.SecretManager/Readme.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        name: dough-provider-token
+        location: reference1
+      # other parameters are just strings, and their interpretation depends on the type of the secret
+      cheese: cheddar
+      sauce: marinara
+      topping: pepperoni
+```
+
+## Secret Types
+Each secret in the manifest has a specified type, and these types determine what parameters are required, and what ends up in the secret store. Some secret types require human interaction to generate and/or rotate. Some secret types also produce multiple values. These values are all stored with additional suffixes in the secret store. Each type is documented below.
```
Comment (garath) [inline src/Microsoft.DncEng.SecretManager/Readme.md:55]: How do I know when human intervention is required?

File: src/Microsoft.DncEng.SecretManager/Readme.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+# encryption keys that are needed in the vault, these keys will be created if they don't exist.
+keys:
+  encryption-key-one:
+    type: RSA
+    size: 2048
+
+# each entry below is a separate secret that will be maintained in the secret store
+# the type identifies what kind of secret it is, and the parameters drive the creation of the secret
+secrets:
+  secret1:
```
Comment (garath) [inline src/Microsoft.DncEng.SecretManager/Readme.md:38]: I think this example would be more useful if it matched a real world situation. Maybe something from our own patterns of use.

File: src/Microsoft.DncEng.SecretManager/Readme.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
+# The .NET Core Engineering Secret Manager
+This tool is used to track and rotate the secrets in one or more key vaults, and to validate the usages of the same. This is done through a yaml manifest file for each vault, and json settings files for each service.
```
Comment (garath) [inline src/Microsoft.DncEng.SecretManager/Readme.md:2]: > This is done through a yaml manifest file for each vault, and json settings files for each service.

I see the vault manifest described below but I don't see anything about the service settings file. Seems like a missing section?

File: src/Microsoft.DncEng.SecretManager/Readme.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+## Rotating secrets
+The synchronize command can be used to rotate secrets that need rotation. It will automatically rotate all secrets that have either expired or hit their rotation time. This command is run as part of a scheduled build to keep all secrets up to date. Any secrets that can be rotated without a human involved will be rotated automatically. For secrets that require human interaction, the build will fail and a human will have to run synchronize.
+
+## Creating new secrets
+New secrets can be created by adding new entries to the manifest file, then running synchronize. The tool will then create the new secret and add it to the secret store.
```
Comment (garath) [inline src/Microsoft.DncEng.SecretManager/Readme.md:8]: > then running synchronize.

Does this mean there is a CLI component? If so, some documentation about its operation would be good. 

Is this what one would do to set up a new service? Consider having a walk-through section for whatever scenarios you expect to be common (like configuring a new service, migrating an existing service to this system).

File: src/Microsoft.DncEng.SecretManager/Readme.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+## Secret Types
+Each secret in the manifest has a specified type, and these types determine what parameters are required, and what ends up in the secret store. Some secret types require human interaction to generate and/or rotate. Some secret types also produce multiple values. These values are all stored with additional suffixes in the secret store. Each type is documented below.
+
+### Azure Storage
+
+#### Connection String
+```yaml
+type: azure-storage-connection-string
+parameters:
+  subscription: azure subscription id
```
Comment (garath) [inline src/Microsoft.DncEng.SecretManager/Readme.md:63]: Do these subscription IDs need to be the GUID, or is the friendly name acceptable?

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 3 above)
Comment (garath) [reply src/Microsoft.DncEng.SecretManager/Readme.md:38]: "SecretReference" is a good example of this as I really don't understand how it is meant to be used. 

What is a "raw string" in this context? Is it the name of a secret that is defined elsewhere in the YAML?

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 1 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md:196]: Ahh yes, I will add it.

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 3 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md:38]: The synchronize command will either fail, or prompt the user if a TTY is detected. I can add that.

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 2 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md:55]: The synchronize command will either fail, or prompt the user if a TTY is detected. I can add that.

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 3 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md:38]: I can do that.

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 4 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md:2]: I didn't add that, I will do so.

**Comment by garath** [comment]
Watch out for capitalization. Products and brands like Azure, GitHub, Service Bus, Event Bus, etc need to be cased correctly when used in text.

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 5 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md:8]: The whole thing is a CLI, any operation will be through a command line operation.

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 6 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md:63]: They must be the guid.

File: src/Microsoft.DncEng.SecretManager/Readme.md (hunk 3 above)
Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md:38]: "raw string" just means "you passed a single string, instead of an object"

Comment (alexperovich) [reply src/Microsoft.DncEng.SecretManager/Readme.md]: I updated the examples. Does it make sense now?

**Review by garath — APPROVED** [review]

**Pull request merged**